### PR TITLE
Add clse_v1.1, which includes parsed linguistic attributes from the linguistic signature, with accompanying script.

### PR DESCRIPTION
**Pull request by justinxzhao**
Additional changes:
- Update gitignore
- Add pre-commit to lint/de-flake python files

### CODE REVIEW

**Comment by justinxzhao** [comment]
CLA check has been signed. :)

File: scripts/generate_v1.1_data.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    "InflectedForm.",
+]
+
+
+def get_linguistic_attribute_dict(linguistic_signature: str) -> Dict[str, str]:
+    """Returns a dictionary of linguistic attribute -> value."""
+    # Strip verbose prefixes from the linguistic signature.
+    reduced_linguistic_signature = linguistic_signature
+    for verbose_signature_prefix in VERBOSE_SIGNATURE_PREFIXES:
+        reduced_linguistic_signature = reduced_linguistic_signature.replace(
```
Comment (varepsilon) [inline scripts/generate_v1.1_data.py]: here you are not checking that those are really prefixes. I suggest you strip the prefixes down below instead (see a comment)

File: scripts/generate_v1.1_data.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+    # Split the signature into constituent parts.
+    attributes = reduced_linguistic_signature.split(",")
+
+    # Parse attributes into a map.
+    attribute_map = {}
+    for attribute in attributes:
+        if not attribute:
+            continue
+        components = attribute.split(":")
```
Comment (varepsilon) [inline scripts/generate_v1.1_data.py]: name, value = split(":", 1)

File: scripts/generate_v1.1_data.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        attribute_map[components[0]] = components[1]
```
Comment (varepsilon) [inline scripts/generate_v1.1_data.py]: for prefix in VERBOSE_SIGNATURE_PREFIXES:
  name = name.removeprefix(prefix)
attribute_map[name] = value

File: scripts/generate_v1.1_data.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+
+def get_all_attributes() -> List[str]:
+    """Returns a sorted list of the superset of all linguistic attributes."""
+    all_attributes = set()
+    with open(CORPUS_PATH) as f:
+        reader = csv.DictReader(f)
+        for row in reader:
+            attribute_map = get_linguistic_attribute_dict(row["linguistic_signature"])
+            all_attributes.update(list(attribute_map.keys()))
```
Comment (varepsilon) [inline scripts/generate_v1.1_data.py]: are you sure you need to create a list and can't update directly from keys()?

File: scripts/generate_v1.1_data.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    return sorted(list(all_attributes))
```
Comment (varepsilon) [inline scripts/generate_v1.1_data.py]: sorted(all_attributes) should work too

**Review by varepsilon — CHANGES_REQUESTED** [review]

File: scripts/generate_v1.1_data.py (hunk 1 above)
Comment (justinxzhao) [reply scripts/generate_v1.1_data.py]: Done.

File: scripts/generate_v1.1_data.py (hunk 3 above)
Comment (justinxzhao) [reply scripts/generate_v1.1_data.py]: Done.

File: scripts/generate_v1.1_data.py (hunk 5 above)
Comment (justinxzhao) [reply scripts/generate_v1.1_data.py]: Done.

File: scripts/generate_v1.1_data.py (hunk 4 above)
Comment (justinxzhao) [reply scripts/generate_v1.1_data.py]: Done.

File: scripts/generate_v1.1_data.py (hunk 2 above)
Comment (justinxzhao) [reply scripts/generate_v1.1_data.py]: Done.

**Review by justinxzhao — COMMENTED** [review]
Thanks for the review!

**Review by varepsilon — APPROVED** [review]
Thanks!

**Pull request merged**